pages/virtualization/docker.md

```diff
@@ -4,30 +4,132 @@ title: Docker
+> dependencies into a standardized unit for software development.
+
+In short, it means that getting people set up with a development
+environment can *potentially* be made dramatically easier. And because
+the same tool can be used to deploy to production websites, it also
+means more [dev/prod parity][].
+
+The reality, at the time of this writing, is a little bit more nuanced.
+
+For one thing, Docker itself has a learning curve, and it's still maturing
+as a product. And until [Docker goes native][], for instance, Mac and
+Windows users will have a bit of a harder time than Linux users.
+
+Aside from that, 18F isn't yet deploying Docker containers in production,
+so the dev/prod parity story also falls short at the moment.
+
+However, despite these limitations, Docker can still make development a
+lot easier; this will only improve as Docker matures, and (hopefully) as
+the government adopts container-based infrastructure in production.
+
+### Definitions and Terms
+
+There are a lot of concepts and tools in the Docker ecosystem, and
+understanding the relationship between them can be quite intimidating at
+first. In this section, we'll introduce you to the bare minimum you'll
+need to know to effectively use Docker at 18F.
+
+**Containers** are the fundamental building block in the Docker ecosystem.
+For the most part, you can think of a container as a very lightweight
+virtual machine. Most web applications are actually a collection of
+containers that are connected in some way. For more details, see
+[What is Docker?][].
+
+Another key component of Docker is the `docker` command-line tool. It's
+actually a fairly low-level piece of software. You likely won't use it
+day-to-day; think of it as "plumbing" that higher-level tools use to
+do awesome things.
+
+**Docker Compose** is one such tool. It's like the porcelain fixture that
+sits atop the plumbing and makes everything a lot more user-friendly. It's
+likely what you'll be using most often as a developer. For more details,
+see the [Docker Compose][] product page.
+
+**Docker Machine** is a tool that most folks won't need to know about once
+[Docker goes native][]. Until then, though, if you're on Windows or OS X,
+the key concept to keep in mind is that Docker Machine is responsible for
+running a linux-based virtual machine behind-the-scenes on your computer.
+Whenever you use `docker` or Docker Compose, your command is actually
+proxied over to the VM, which does all the hard work. Also, when you
+run a web server in a Docker container, it'll be exposed on this VM,
+*not* your localhost, which makes figuring out its URL a bit inconvenient.
+More on that later.
 
 ### Troubleshooting
 
-#### Cannot connect to the Docker daemon. Is the docker daemon running on this host?
+Almost all troubleshooting of docker stems from Docker Machine. Again,
+once [Docker goes native][], life will be much easier for OS X and
+Windows users.
+
+#### Connecting To A Container
```

Comment:
You've done a great job of bringing up exactly the stuff that has given me headaches.

Using Kitematic has been the easiest way for me to avoid them - is that worth mentioning?

Then again, I've got my native Docker Beta now - only about 12 hours after requesting, and it will probably not be long before every interested 18Fer does, too.  Maybe the advice should just be "Quit messing around with the VM-based Docker".